### ISSUE TITLE
Some more documentation needed

### ISSUE THREAD

**Issue by burgerga** [opnumten/single_cell_segmentation#2]
Thanks for sharing your code!
I am trying to reproduce the results for T47D and then adapt the code to my own images, but I am missing some pieces of information:

- How to generate the training data? Currently, I do the following:
    * Do manual segmentation of parts of the images with sizes 128x128, 256x256, 512x512 (how many do I need?)
    * Compute distance transform on the manual segmentation
- Do I need to provide bounding box training data for the `deep_cell_detector`? My understanding is that the `deep_cell_detector` makes bounding boxes from the distance transforms, but this bould be the same for different types of images, right? If not, could you include instructions on how to modify `deep_cell_detector/keras_frcnn/config.py`, since that is unclear to me,  `frcnn_apply.ipynb` has some hardcode values for the nmumg dataset.

**Comment by opnumten** [comment]
No, the training patch is randomly cropped from several big images. The distance transform is calculated with scipy.ndimage.morphology.distance_transform_edt.  I provide the training data of deep cell detector in the folder of trainning data/frcnn/Annotations. Yes, it should be the similar. But different cells may have different size. More details about the method could be found in https://arxiv.org/abs/1803.10829.
Config.py are some parameters used in frcnn. What do you mean hardcode values?

**Comment by burgerga** [comment]
Thanks for your answer, after more careful reading I have a better understanding of what the different steps are supposed to do.
Regarding the Config.py, I saw that there were some cell line specific config options, for example in https://github.com/opnumten/single_cell_segmentation/blob/master/T47D_fluorescence/training_data/frcnn/T47D%20parameter%20for%20frcnn%20config and I was wondering how we should determine those for our own data. However, at https://github.com/yhenon/keras-frcnn/ they suggest not using the frcnn anymore, and using https://github.com/fizyr/keras-retinanet. By the way, what is the benefit of using the distance transform in the frcnn and not the raw images?

**Comment by opnumten** [comment]
The parameters are determined by your cell size and cell shape. You could do simple statistics on your training data and obtain the parameters. 
The reason we use frcnn is for generating seeds for watershed. We only want to detect single cell roughly, which do not have a strict criteria on bounding box location accuracy. This step could omitted if you have nucleus channel in your data.  I use frcnn because it is a popular method and I don't care too much about speed. You could try retinanet by yourself in the second step.

Frcnn on Distance transform is much easier to train. To train on the raw images, you need more trainning data which require a lot of labor and time to prepare.

**Comment by burgerga** [comment]
Thanks again!

> This step could omitted if you have nucleus channel in your data.

We have nucleur staining (Hoechst) so we did an Otsu threshold, but then the results from step 3 are not  so great. Even if we use a pixel classification ML approach (Ilastik) for our bounding boxes, the results are slightly better but still not good enough. So I guess we have difficult data, or we need improvement in step 1... or we actually need to do step 2. 

> Frcnn on Distance transform is much easier to train. To train on the raw images, you need more trainning data which require a lot of labor and time to prepare.

Ah that makes sense, creating the training data indeed takes up a lot of time!
